policybot/pkg/testresults/testresults.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Take in a pr number from blob storage and examines the pr
+// for all tests that are run and their results. The results are then written to storage.
+
+package testresults
+
+import (
+	"context"
+	"encoding/json"
+
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	store "istio.io/bots/policybot/pkg/storage"
+	"istio.io/pkg/log"
+)
+
+// Pull struct for the structure under refs/pulls in clone-records.json
+type pull struct {
+	Number int
+	Author string
+	Sha    string
+}
+
+// Cmd struct for Commands object under clone-records.json
+type cmnd struct {
+	Command string
+	Output  string
+}
+
+// Finished struct to store values for all fields in Finished.json
+type finished struct {
+	Timestamp int64
+	Passed    bool
+	Result    string
+}
+
+// Clone_Record struct to store values for all fields in clone-records.json
+type cloneRecord struct {
+	Refs struct {
+		Org       string
+		Repo      string
+		BaseRef   string `json:"base_ref"`
+		BaseSha   string `json:"base_sha"`
+		Pulls     []pull
+		PathAlias string
+	}
+	Commands []cmnd
+	Failed   bool
+}
+
+// Started struct to store values from started.json
+type started struct {
+	Timestamp int64
+}
+
+type PrResultTester struct {
+	Client     *storage.Client
+	ctx        context.Context
```

Comment:
I made a change a while back to remove the use of context instance in the constructors, since that was incorrect. Context objects need to be obtained when a request arrives at the server and passed down to all calls from there.

So the server is called in the webhook handler and that's where the context should come from, and be passed down to the CheckResult function.